pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -4357,39 +4357,50 @@ private PersistentReplicator getReplicatorReference(String replName, PersistentT
         }
     }
 
+    private CompletableFuture<Void> updatePartitionedTopicMetadataAsync(int numPartitions) {
+        CompletableFuture<Void> future = namespaceResources().getPartitionedTopicResources()
+                .updatePartitionedTopicAsync(topicName, p -> new PartitionedTopicMetadata(numPartitions, p.properties));
+        future.exceptionally(ex -> {
+            // If the update operation fails, clean up the partitions that were created
+            getPartitionedTopicMetadataAsync(topicName, false, false).thenAccept(metadata -> {
+                int oldPartition = metadata.partitions;
+                for (int i = oldPartition; i < numPartitions; i++) {
+                    topicResources().deletePersistentTopicAsync(topicName.getPartition(i)).exceptionally(ex1 -> {
+                        log.warn("[{}] Failed to clean up managedLedger {}", clientAppId(), topicName,
+                                ex1.getCause());
+                        return null;
+                    });
+                }
+            }).exceptionally(e -> {
+                log.warn("[{}] Failed to clean up managedLedger", topicName, e);
+                return null;
+            });
+            return null;
+        });
+        return future;
+    }
 
     private CompletableFuture<Void> updatePartitionedTopic(TopicName topicName, int numPartitions, boolean force) {
         CompletableFuture<Void> result = new CompletableFuture<>();
-        createSubscriptions(topicName, numPartitions).thenCompose(__ -> {
-            CompletableFuture<Void> future = namespaceResources().getPartitionedTopicResources()
-                    .updatePartitionedTopicAsync(topicName, p ->
-                        new PartitionedTopicMetadata(numPartitions, p.properties));
-            future.exceptionally(ex -> {
-                // If the update operation fails, clean up the partitions that were created
-                getPartitionedTopicMetadataAsync(topicName, false, false).thenAccept(metadata -> {
-                    int oldPartition = metadata.partitions;
-                    for (int i = oldPartition; i < numPartitions; i++) {
-                        topicResources().deletePersistentTopicAsync(topicName.getPartition(i)).exceptionally(ex1 -> {
-                            log.warn("[{}] Failed to clean up managedLedger {}", clientAppId(), topicName,
-                                    ex1.getCause());
+        createSubscriptions(topicName, numPartitions).thenCompose(
+                        __ -> updatePartitionedTopicMetadataAsync(numPartitions))
+                .thenAccept(__ -> result.complete(null))
+                .exceptionally(ex -> {
+                    if (force && ex.getCause() instanceof PulsarAdminException.ConflictException) {
+                        log.warn("[{}] update partitioned topic's partition num {} with ignoring conflict exception",
```

Comment:
This exception is throw from `createSubscriptions`, I think it's better handle the exception in `createSubscriptions`. BTW,  could we make sure all the subscriptions have been created if `ConflictException` happens and we ignore it here.

Reply:
1. I don't think so. The Conflict exception can ignore only when `force` is true. It is better to keep as it is .
> it's better handle the exception in createSubscriptions

2. The `createSubscriptions` can make sure all the subscriptions are created because of the internal `FutureUtil.waitForAll` .

Reply:
> The createSubscriptions can make sure all the subscriptions are created because of the internal FutureUtil.waitForAll .

What if one of the subscriptions throw a non  ConflictException,  for example:

- There are 100 subs in the topic.
- When 'createSubscriptions',  the 1st future throws `ConflictException`, and others throw non `ConflictException` , we will miss creating some subscriptions.

So we'd better handle the `ConflictException` in `createSubscriptions()` method @HQebupt